modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -1091,7 +1179,7 @@ class ResizeSubgraph3 : public Subgraph
         int cast = addNodeToMatch("Cast", concat1);
 
         int shape2 = addNodeToMatch("Shape", input);
-        int slice = addNodeToMatch("Slice", shape2, addNodeToMatch("Constant"), addNodeToMatch("Constant"), addNodeToMatch("Constant"));
+        int slice = addNodeToMatch("Slice", shape2, addNodeToMatch("Constant"), addNodeToMatch("Constant"));
```

Comment:
Should `ResizeSubgraph1` be modified in the same way?

Reply:
No. This subgraph is added to simplify model `resize_nearest_unfused_opset11_torch1.4.onnx` in opencv_extra. This graph does not have shape pre-inferred, so there is no value info proto for the input of the Slice operator. The actual `axes` value is 0, which is not -1 and cannot be recognized as the last axis because there is no shape inference information.

Reply:
Got it. So as I tried propose [here](https://github.com/opencv/opencv/pull/24655#discussion_r1416744142) can we align all the Slice layers matches to accept 5 mandatory inputs but add a pattern which appends optional inputs if missed?

Reply:
However, current proposal also looks good to me and it make sense to merge to unblock Attention layer.

Reply:
> 5 inputs

I would say it can be more robust as it also handles cases where value info proto is not available. However, it needs to create ONNX Constant nodes for axes and steps in finalize() which I am not sure how to do.